Library/Homebrew/cask/cask_loader.rb

```diff
@@ -322,12 +322,14 @@ def from_h_hash_gsubs(hash)
       end
 
       def from_h_gsubs(value)
+        return value if value.blank?
+
         if value.respond_to? :to_h
           from_h_hash_gsubs(value)
         elsif value.respond_to? :to_a
           from_h_array_gsubs(value)
         else
-          from_h_string_gsubs(value)
+          { "true" => true, "false" => false }.fetch(value, from_h_string_gsubs(value))
```

Comment:
Is this just a temporary workaround? I just checked on `cask.json` file after this got merged in and it doesn't seem to include any `"true"` or `"false"` strings anymore.

Reply:
@apainintheneck Can you elaborate a bit for @Rylan12 what you expected the behaviour to be, whether it would be good/bad for this to be desirable and what problems you're setting now vs. before? Just not totally sure I understand these from this comment alone 🙇🏻

Reply:
I was saying that since this PR seems to fix the issue where bools were being turned into strings when generating the JSON for a cask do we still need this workaround for bools that are strings when loading casks from the API?

The first part of this shows that the JSON for the `finch` cask now includes `"sudo"=>true` so no conversion from `"true"` to `true` is probably necessary anymore.

```rb
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
irb(main):001:0> cask_json = Homebrew::API::Cask.all_casks["finch"]; nil
=> nil
irb(main):002:0> cask_json["artifacts"]
=> [{"uninstall"=>[{"script"=>{"executable"=>"/Applications/Finch/uninstall.sh", "sudo"=>true}}]}, {"pkg"=>["Finch-v0.3.0-x86_64.pkg"]}, {"zap"=>[{"trash"=>"~/.finch"}]}]
```

This second part shows the output of running the `.from_h_hash_gsubs` method on the artifacts for that cask which mimics the following line in the `FromAPILoader`.

https://github.com/Homebrew/brew/blob/b3a71ba7cdb7e005c26c86bcc627f0549dfe19d0/Library/Homebrew/cask/cask_loader.rb#L234

It turns the bool back into a string again.  Is this expected behavior?

```rb
irb(main):003:0> loader = Cask::CaskLoader::FromAPILoader.new("")
=> #<Cask::CaskLoader::FromAPILoader:0x00007fd17581c510 @token="", @path=#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/.rb>, @from_json=nil>
irb(main):004:0> cask_json["artifacts"].map { |art| loader.send(:from_h_hash_gsubs, art) }
=> [{"uninstall"=>[{"script"=>{"executable"=>"/Applications/Finch/uninstall.sh", "sudo"=>"true"}}]}, {"pkg"=>["Finch-v0.3.0-x86_64.pkg"]}, {"zap"=>[{"trash"=>"~/.finch"}]}]
irb(main):005:0> exit
```